restli-server/src/main/java/com/linkedin/restli/internal/server/response/ResponseUtils.java

```diff
@@ -98,6 +189,15 @@ public static RestResponse buildResponse(RoutingResult routingResult, RestLiResp
     if (restLiResponse.hasData() && ResourceEntityType.STRUCTURED_DATA == resourceEntityType)
     {
       DataMap dataMap = restLiResponse.getDataMap();
+      if (context.isDefaultValueFillInRequested())
+      {
+        Class<? extends RecordTemplate> valueClass = routingResult.getResourceMethod().getResourceModel().getValueClass();
+        if (valueClass != null)
+        {
+          DataSchema dataSchema = DataTemplateUtil.getSchema(valueClass);
+          dataMap = fillInDefaultValues(dataSchema, dataMap);
```

Comment:
I'm don't think this would work for batch, finder or action methods. Can you add test coverage for these?

Check the restli validator on response method to understand different response structures. I feel it might be easier to add the fill default logic in ResponseBuilders

Reply:
I saw there is a `RestLiValidationFilter` which implements `Filter` interface, it has a onResponse function, 
also there are multiple `**ResponseBuilders` where they all share same interface `RestLiResponseBuilder`
I don't know how to fit the `fill-default` in there yet, will dig deeper.